core/src/test/java/org/jboss/set/mjolnir/archive/mail/report/InvalidResponsiblePersonTableTestCase.java

```diff
@@ -0,0 +1,96 @@
+import org.jsoup.select.Elements;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.inject.Inject;
+import javax.naming.NamingException;
+import javax.persistence.EntityManager;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doReturn;
+
+@RunWith(CdiTestRunner.class)
+public class InvalidResponsiblePersonTableTestCase {
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(8089);
+
+    @Inject
+    private EntityManager em;
+
+    @Inject
+    private LdapScanningBean ldapScanningBean;
+
+    @Inject
+    private InvalidResponsiblePersonTable invalidResponsiblePersonTable;
+
+    @Inject
+    private LdapDiscoveryBean ldapDiscoveryBeanMock;
+
+    @Before
+    public void setup() throws NamingException {
+
+        HashMap<Object, Object> ldapUsersMap = new HashMap<>();
+        ldapUsersMap.put("Responsible guy", false);
+        ldapUsersMap.put("Responsible guy1", false);
+        ldapUsersMap.put("non-existing", true);
+        doReturn(ldapUsersMap).when(ldapDiscoveryBeanMock).checkUsersExists(ArraysUtils.asSet("Responsible guy", "Responsible guy1", "non-existing"));
+        em.getTransaction().begin();
+
+        RegisteredUser registeredUser = new RegisteredUser();
+        registeredUser.setGithubName("bruno");
+        registeredUser.setKerberosName("bruno");
+        registeredUser.setResponsiblePerson("Responsible guy");
+        em.persist(registeredUser);
+
+        registeredUser = new RegisteredUser();
+        registeredUser.setGithubName("joe");
+        registeredUser.setKerberosName("joe");
+        registeredUser.setResponsiblePerson("Responsible guy1");
+        em.persist(registeredUser);
+
+        registeredUser = new RegisteredUser();
+        registeredUser.setGithubName("sam");
+        registeredUser.setKerberosName("samExisting");
+        registeredUser.setResponsiblePerson("non-existing");
+        registeredUser.setWhitelisted(true);
+        em.persist(registeredUser);
+
+        em.getTransaction().commit();
+
+    }
+
+    @Test
+    public void testComposeTableBody() throws NamingException, IOException {
+
+        List<RegisteredUser> listOfInvalidResponsiblePerson = invalidResponsiblePersonTable.getInvalidLdapUser();
+        assertThat(listOfInvalidResponsiblePerson).extracting(RegisteredUser::getKerberosName).containsOnly("joe", "bruno");
+
+        String messageBody = invalidResponsiblePersonTable.composeTable();
+        Document doc = Jsoup.parse(messageBody);
+
+        assertThat(doc.select("tr").size()).isEqualTo(listOfInvalidResponsiblePerson.size() + 1);
+        assertThat(doc.select("th").text()).isEqualTo("Name Responsible person");
+
+        Elements elements = doc.select("td");
+        assertThat(elements.size()).isEqualTo(listOfInvalidResponsiblePerson.size() * 2);
```

Comment:
Should we check here that the invalid user's names are present in the table? Something like:

```
assertThat(elements.get(0).text()).contains("Responsible guy");
assertThat(elements.get(1).text()).contains("Responsible guy 1");
```

or something like that.